### PR TITLE
k3s: add support for port-forwarding with IMAGETEST_PORT_FORWARDS

### PR DESCRIPTION
**Pull request by Dentrax**
This PR introduces the capability to specify multiple port mappings for the k3s harness running as containers. This allows users to define local-to-remote port forwarding in the format using the brand-new introduced `$IMAGETEST_PORT_FORWARDS` environment variable with format of `LOCAL_PORT:DEST_PORT`, enabling access to ports running inside the Docker container from the local environment.

If `:` not provided, the target port will be same as the local port.

**Example:**
```
export IMAGETEST_PORT_FORWARDS="8065,9090:57650"
```

### CODE REVIEW

File: internal/containers/provider/docker.go [hunk 1]
```diff
@@ -51,6 +52,10 @@ type DockerRequest struct {
 	// ManagedVolumes is the list of volumes that should be torn down when the
 	// provider finishes execution
 	ManagedVolumes []mount.Mount
+	// PortBindings is the list of port bindings to apply to the container
+	// when it is created. The key is the port to bind to on the host, and the
+	// value is the port to bind to on the container.
+	PortBindings map[nat.Port][]nat.PortBinding
```
Comment (stormqueen1990) [inline internal/containers/provider/docker.go:58]: We need to add a new option here for exposed ports. The type is `nat.PortSet`: https://github.com/moby/moby/blob/c2879978022db54ef21ac93feeabce0cbd75f63d/api/types/container/config.go#L51
This option will need to be plumbed through the `container.Config`, in the `ExposedPorts` field: https://github.com/chainguard-dev/terraform-provider-imagetest/blob/765d75d50bd669e49cdea767d356ece1cfd45525/internal/containers/provider/docker.go#L158-L167

We could also use the predefined type `nat.PortMap` for the `PortBindings` variable:
```suggestion
	PortBindings nat.PortMap
```

File: internal/harnesses/k3s/k3s.go [hunk 2]
```diff
@@ -94,6 +96,28 @@ func New(id string, cli *provider.DockerClient, opts ...Option) (types.Harness,
...
+		portBindings[targetPort] = []nat.PortBinding{
+			{
+				HostIP:   "0.0.0.0",
+				HostPort: localPort,
+			},
+		}
+	}
+
+	harnessOptions.Sandbox.PortBindings = portBindings
+
```
Comment (stormqueen1990) [inline internal/harnesses/k3s/k3s.go:120]: This can be simplified by using the functions that the Docker SDK provides. We will also need the exposed ports output of that function:
```suggestion
    portMappings := make([]string, 0)
	for _, mapping := range harnessOptions.PortMappings {
		// If no `:` provided, assume the target port will be same as the local port
		if !strings.Contains(mapping, ":") {
			mapping = mapping + ":" + mapping
		}
		
		portMappings = append(portMappings, mapping)
	}
	
	exposedPorts, portBindings, err := nat.ParsePortSpecs(portMappings)
	if err != nil {
	    // error handling here
	}

	harnessOptions.Sandbox.PortBindings = portBindings
	harnessOptions.Sandbox.ExposedPorts = exposedPorts
```

File: internal/provider/store.go [hunk 3]
```diff
@@ -118,6 +119,12 @@ func (s *ProviderStore) SkipTeardown() bool {
 	return v != ""
 }
 
+// PortForwards returns the IMAGETEST_PORT_FORWARDS environment variable.
+func (s *ProviderStore) PortForwards() []string {
+	v := os.Getenv("IMAGETEST_PORT_FORWARDS")
```
Comment (stormqueen1990) [inline internal/provider/store.go:124]: This has the potential of causing trouble if the empty string isn't handled. Suggestion:
```suggestion
	v, ok := os.LookupEnv("IMAGETEST_PORT_FORWARDS")
	if !ok || "" == v {
	    return nil
	}
```

**Comment by Dentrax** [comment]
Thank you so much for your reviews @stormqueen1990! Is there any testing steps that I can follow to test the functionality?

File: .golangci.yml
Comment (stormqueen1990) [inline .golangci.yml:1]: Can we move this change into a separate PR since it's unrelated?

**Comment by stormqueen1990** [comment]
> Thank you so much for your reviews @stormqueen1990! Is there any testing steps that I can follow to test the functionality?

`kubectl port-forward` perhaps, with a `curl` command to try and access the exposed port?

**Comment by Dentrax** [comment]
The functionality is now merged on https://github.com/chainguard-dev/terraform-provider-imagetest/pull/132!
